pkg/sippyserver/server.go

```diff
@@ -1286,6 +1333,12 @@ func (s *Server) Serve() {
 			Capabilities: []string{LocalDBCapability},
 			HandlerFunc:  s.jsonJobBugsFromDB,
 		},
+		{
+			EndpointPath: "/api/jobs/artifacts",
+			Description:  "Queries job artifacts and their contents",
+			Capabilities: []string{LocalDBCapability},
+			HandlerFunc:  s.queryJobArtifacts,
+		},
```

Comment:
Should this be cached? If we share the links around, it might be nice if the response was immediate on subsequent queries. Just needs a CacheTime entry to turn it on

Reply:
I don't like the idea of caching at this level given there's some possibility of returning  transient error responses among otherwise successful content.

My experience is that responses are pretty quick already (a few seconds even with quite a lot of results), although that's just in testing against my own system.

Also I suspect most usage of this API will be indirect via the UI, where the user may be tweaking the set of job runs included; it may make more sense to cache responses at the individual job run level (and filter out any that saw errors).

I think we can wait until performance becomes problematic and decide how to cache then based on actual usage.